src/main/java/homer/api/DeviceView.java

```diff
@@ -5,18 +5,18 @@
  * 
  * @param <S> The device state.
  */
-public interface DeviceView<S> {
+public interface DeviceView {
 
     /**
      * 
      * @return Device information to be displayed visually.
      */
-    S getState();
+    DeviceState getState();
```

Comment:
Should we really have a `getState()` in the view? If I understood mvc correctly, this view should only have the `setState()` that is called by the controller

Reply:
>Should we really have a getState() in the view? If I understood mvc correctly, this view should only have the setState() that is called by the controller

how do we deal with incoming updates then?

Reply:
actually, i think you're right